src/transformers/models/bark/modeling_bark.py

```diff
@@ -0,0 +1,1429 @@
+            x_semantic_history = np.repeat(
+                history_prompt["semantic_prompt"][None], batch_size, axis=0
+            )  # TODO: already used before
+            x_coarse_history = history_prompt["coarse_prompt"]
+
+            x_coarse_history = (
+                _flatten_codebooks(x_coarse_history, self.config.codebook_size) + self.config.semantic_vocab_size
+            )
+            x_coarse_history = np.repeat(x_coarse_history[None], batch_size, axis=0)
+            # e.g: after SEMANTIC_VOCAB_SIZE (10000), 1024 tokens dedicated to first codebook, 1024 next tokens dedicated to second codebook.
+
+            max_semantic_history = int(np.floor(max_coarse_history / semantic_to_coarse_ratio))
+            # trim histories correctly
+            n_semantic_hist_provided = np.min(
+                [
+                    max_semantic_history,
+                    x_semantic_history.shape[1] - x_semantic_history.shape[1] % 2,
+                    int(np.floor(x_coarse_history.shape[1] / semantic_to_coarse_ratio)),
+                ]
+            )
+
+            n_coarse_hist_provided = int(round(n_semantic_hist_provided * semantic_to_coarse_ratio))
+
+            x_semantic_history = x_semantic_history[:, -n_semantic_hist_provided:].astype(np.int32)
+            x_coarse_history = x_coarse_history[:, -n_coarse_hist_provided:].astype(np.int32)
+            # bit of a hack for time alignment (sounds better) - from Bark original implementation
+            x_coarse_history = x_coarse_history[:, :-2]
+
+        else:
+            # shape: (batch_size, 0)
+            x_semantic_history = np.array([[]] * batch_size, dtype=np.int32)
+            x_coarse_history = np.array([[]] * batch_size, dtype=np.int32)
+
+        x_semantic_history = torch.from_numpy(x_semantic_history).to(self.device)
+        x_coarse_history = torch.from_numpy(x_coarse_history).to(self.device)
+
+        return x_semantic_history, x_coarse_history
+
+    def generate_text_semantic(
+        self,
+        input_ids: torch.Tensor,
+        history_prompt: Optional[Dict[str, np.ndarray]] = None,
+        **kwargs,
+    ) -> torch.LongTensor:
+        # TODO: add a max_gen_duration_s early stop
+
+        # TODO: Not used for now. where to set the default value ?
+        # min_eos_p = kwargs.get("min_eos_p", 0.2)
+
+        # TODO: input_ids[:,256:256+256] (to verify) corresponds to history_prompt["semantic_prompt"], maybe use that
+        # input_ids should be of shape (batch_size, seq_len) where seq_len = 513
+        batch_size = input_ids.shape[0]
+
+        input_ids = input_ids + self.config.text_encoding_offset
+
+        if kwargs.get("attention_mask", None) is not None:
+            input_ids.masked_fill_((1 - kwargs.pop("attention_mask")).bool(), self.config.text_pad_token)
+
+        if history_prompt is not None:
+            semantic_history = history_prompt["semantic_prompt"][-256:]
+            semantic_history = np.pad(
```

Comment:
Should be in `torch` here - the semantic history should be passed as a torch tensor if we set `return_tensors="pt"` when we call the processor